src/duct/core/repl.clj

```diff
@@ -17,3 +19,31 @@
   fipp/IOverride
   fipp/IEdn
   (-edn [r] (tagged-literal 'duct/resource (:path r))))
+
+(def ^:private event->file-name
+  (comp #(.getName %)
+        :file))
+
+(def ^:private matches-clojure-file?
+  (partial re-matches #"^[a-zA-Z]{1}[a-zA-Z_]*(.clj|.edn)$"))
+
+(defn- clojure-file?
+  [ctx event]
+  (-> event
+      event->file-name
+      matches-clojure-file?))
+
+(defn- auto-reset-handler
+  [ctx event]
+  (binding [*ns* true]
```

Comment:
I don't understand why you're binding `*ns*` to `true` here. Can you explain?

Reply:
With the following code:

``` Clojure
(defn- auto-reset-handler
  [ctx event]
  (require 'integrant.repl)
  (integrant.repl/reset)
  ctx)
```

I get this error:

```
java.lang.IllegalStateException: Can't change/establish root binding of: *ns* with set
	at clojure.lang.Var.set(Var.java:226)
	at clojure.lang.RT$1.invoke(RT.java:244)
	at clojure.tools.namespace.repl$do_refresh.invokeStatic(repl.clj:95)
	at clojure.tools.namespace.repl$do_refresh.invoke(repl.clj:82)
	at clojure.tools.namespace.repl$refresh.invokeStatic(repl.clj:145)
	at clojure.tools.namespace.repl$refresh.doInvoke(repl.clj:128)
	at clojure.lang.RestFn.invoke(RestFn.java:421)
	at integrant.repl$reset.invokeStatic(repl.clj:84)
	at integrant.repl$reset.invoke(repl.clj:82)
	at duct.core.repl$auto_reset_handler.invokeStatic(form-init3993148749624374756.clj:42)
	at duct.core.repl$auto_reset_handler.invoke(form-init3993148749624374756.clj:36)
	at hawk.core$process_handler$fn__23908.invoke(core.clj:41)
	at clojure.lang.AFn.applyToHelper(AFn.java:156)
	at clojure.lang.AFn.applyTo(AFn.java:144)
	at clojure.core$apply.invokeStatic(core.clj:665)
	at clojure.core$apply.invoke(core.clj:660)
	at hawk.core$catch_errors$fn__23881.doInvoke(core.clj:8)
	at clojure.lang.RestFn.invoke(RestFn.java:421)
	at clojure.core$binding_conveyor_fn$fn__5739.invoke(core.clj:2036)
	at clojure.lang.AFn.applyToHelper(AFn.java:156)
	at clojure.lang.RestFn.applyTo(RestFn.java:132)
	at clojure.lang.Agent$Action.doRun(Agent.java:114)
	at clojure.lang.Agent$Action.run(Agent.java:163)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

Binding `*ns*` to true and then using `in-ns` _somehow_ fixes this. As stated in my PR description I'm not happy about this solution and would like to find a cleaner one. So far I've been unable to do so sadly.

Reply:
What about:

```clojure
(defn- auto-reset-handler [ctx event]
  (binding [*ns* *ns*]
    (integrant.repl/reset))
  ctx)
```

I suspect this is caused by Hawk calling the handler in a separate thread.

Reply:
That seems to work